### ISSUE TITLE
[clang-tidy] readability-trailing-comma false positive on empty initializer list

### ISSUE THREAD

**Issue by tom-anders** [llvm/llvm-project#220222]
```c++
struct S {
    int foo = 1;
};

void bug(S, int);

int main() {
    bug(S{}, 1);
}
```

triggers a false positive warning:

```
<source>:8:12: warning: initializer list should not have a trailing comma [readability-trailing-comma]
    8 |     bug(S{}, 1);
      |            ^
1 warning generated.
```

https://godbolt.org/z/zMnz43bn3

This warning disappears if `S` has no fields or if the second parameter of the `bug` function is removed

Might be related to https://github.com/llvm/llvm-project/issues/183517
